mmcv/ops/csrc/tensorrt/plugins/trt_nms.cpp

```diff
@@ -0,0 +1,252 @@
+    const nvinfer1::PluginTensorDesc *outputs, int nbOutputs) const {
+  size_t boxes_word_size = mmcv::getElementSize(inputs[0].type);
+  size_t num_batches = inputs[0].dims.d[0];
+  size_t spatial_dimension = inputs[0].dims.d[1];
+  size_t num_classes = inputs[1].dims.d[1];
+
+  return get_onnxnms_workspace_size(num_batches, spatial_dimension, num_classes,
+                                    boxes_word_size, mCenterPointBox);
+}
+
+int ONNXNonMaxSuppressionDynamic::enqueue(
+    const nvinfer1::PluginTensorDesc *inputDesc,
+    const nvinfer1::PluginTensorDesc *outputDesc, const void *const *inputs,
+    void *const *outputs, void *workSpace, cudaStream_t stream) {
+  int num_batches = inputDesc[0].dims.d[0];
+  int spatial_dimension = inputDesc[0].dims.d[1];
+  int num_classes = inputDesc[1].dims.d[1];
+
+  const float *boxes = (const float *)inputs[0];
+  const float *scores = (const float *)inputs[1];
+  const int *max_output_boxes_per_class =
+      (mNumberInputs >= 3) ? (const int *)inputs[2] : nullptr;
+  const float *iou_threshold =
+      (mNumberInputs >= 4) ? (const float *)inputs[3] : nullptr;
+  const float *score_threshold =
+      (mNumberInputs >= 5) ? (const float *)inputs[4] : nullptr;
+
+  int *output = (int *)outputs[0];
+
+  TRTONNXNMSCUDAKernelLauncher_float(
+      boxes, scores, max_output_boxes_per_class, iou_threshold, score_threshold,
+      output, mCenterPointBox, num_batches, spatial_dimension, num_classes,
+      workSpace, stream);
+
+  return 0;
+}
+
+nvinfer1::DataType ONNXNonMaxSuppressionDynamic::getOutputDataType(
+    int index, const nvinfer1::DataType *inputTypes, int nbInputs) const {
+  return nvinfer1::DataType::kINT32;
+}
+
+// IPluginV2 Methods
+const char *ONNXNonMaxSuppressionDynamic::getPluginType() const {
+  return PLUGIN_NAME;
+}
+
+const char *ONNXNonMaxSuppressionDynamic::getPluginVersion() const {
+  return PLUGIN_VERSION;
+}
+
+int ONNXNonMaxSuppressionDynamic::getNbOutputs() const { return 1; }
+
+int ONNXNonMaxSuppressionDynamic::initialize() { return 0; }
+
+void ONNXNonMaxSuppressionDynamic::terminate() {}
+
+size_t ONNXNonMaxSuppressionDynamic::getSerializationSize() const {
+  return sizeof(mCenterPointBox);
+}
+
+void ONNXNonMaxSuppressionDynamic::serialize(void *buffer) const {
+  serialize_value(&buffer, mCenterPointBox);
+}
+
+void ONNXNonMaxSuppressionDynamic::destroy() {
+  // This gets called when the network containing plugin is destroyed
+  delete this;
+}
+
+void ONNXNonMaxSuppressionDynamic::setPluginNamespace(
+    const char *libNamespace) {
+  mNamespace = libNamespace;
+}
+
+const char *ONNXNonMaxSuppressionDynamic::getPluginNamespace() const {
+  return mNamespace.c_str();
+}
+
+////////////////////// creator /////////////////////////////
+
+ONNXNonMaxSuppressionDynamicCreator::ONNXNonMaxSuppressionDynamicCreator() {
+  mPluginAttributes.clear();
+  // mPluginAttributes.emplace_back(nvinfer1::PluginField("center_point_box"));
```

Comment:
Could this line be removed if it is not a comment?

Reply:
There is an optional attribute [center_point_box](https://github.com/onnx/onnx/blob/master/docs/Changelog.md#attributes-138) in onnx nms.
TensorRT can not parse the attribute if it is not explicit in the model. with error log:
```
IndexError: Attribute not found: center_point_box
```
That's why I comment this line to disable this attribute. Should I add a TODO here?

Reply:
Is there a way to solve this? say, set a default attribute?

Reply:
Yes, I have already add the default attribute to the node when preprocess onnx.